java/tools/src/test/java/org/apache/arrow/tools/TestIpcFuzz.java

```diff
@@ -0,0 +1,125 @@
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.ipc.ArrowFileReader;
+import org.apache.arrow.vector.ipc.ArrowStreamReader;
+import org.apache.arrow.vector.util.ValueVectorUtility;
+import org.apache.commons.io.filefilter.WildcardFileFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TestIpcFuzz {
+
+  static final List<String> WHITE_LIST = new ArrayList<>();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TestIpcFuzz.class);
+
+  static {
+    WHITE_LIST.add("clusterfuzz-testcase-minimized-arrow-ipc-file-fuzz-5707423356813312");
+  }
+
+  static void readIpcFile(File ipcFile) {
+    BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE);
+    try (ArrowFileReader reader = new ArrowFileReader(new FileInputStream(ipcFile).getChannel(), allocator)) {
+      VectorSchemaRoot root = reader.getVectorSchemaRoot();
+
+      // validate schema
+      ValueVectorUtility.validate(root);
+
+      while (reader.loadNextBatch()) {
+        ValueVectorUtility.validateFull(root);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  static void readIpcStream(File ipcFile) {
+    BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE);
+    try (ArrowStreamReader reader = new ArrowStreamReader(new FileInputStream(ipcFile).getChannel(), allocator)) {
+      VectorSchemaRoot root = reader.getVectorSchemaRoot();
+
+      // validate schema
+      ValueVectorUtility.validate(root);
+
+      while (reader.loadNextBatch()) {
+        ValueVectorUtility.validateFull(root);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  static File[] getTestFiles(String testFilePath) {
+    int idx = testFilePath.lastIndexOf(File.separator);
+    File directory = new File(testFilePath.substring(0, idx));
+    String filter = testFilePath.substring(idx + 1);
+    FileFilter fileFilter = new WildcardFileFilter(filter);
+    return directory.listFiles(fileFilter);
+  }
+
+  static void testFuzz(boolean stream, File test) {
+    if (stream) {
+      readIpcStream(test);
+    } else {
+      readIpcFile(test);
+    }
+  }
+
+  public static void main(String[] args) {
+    if (args.length < 2) {
+      LOGGER.error("Usage: <cmd> [stream|file] <test file path>");
+      System.exit(1);
+    }
+
+    final boolean stream;
+    if (args[0].equalsIgnoreCase("file")) {
+      stream = false;
+    } else if (args[0].equalsIgnoreCase("stream")) {
+      stream = true;
+    } else {
+      throw new IllegalArgumentException("The first argument must be file or stream");
+    }
+
+    File[] testFiles = getTestFiles(args[1]);
+    for (File test : testFiles) {
+      LOGGER.info("Testing file " + test.getName());
+      if (WHITE_LIST.contains(test.getName())) {
+        testFuzz(stream, test);
+        LOGGER.info("Test finished successfully.");
+      } else {
+        Exception e = assertThrows(Exception.class,
```

Comment:
wouldn't we want a specific exception here?  Does this have a non-zero exit code if it failes?

Reply:
> wouldn't we want a specific exception here? 
I am afraid there is not a specific exception here, because some exception are thrown when validating the vector (ValidateException), while others may be thown when creating the schema/vector (maybe IllegalArgumentException)

> Does this have a non-zero exit code if it failes?
Yes.